### PR TITLE
SONARJS-661 DeadStore: false negative with logical operator in assignment

### PR DESCRIPTION
**Pull request by pynicolas**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [This branch's code block is the same as the block for the branch on line 228.](https://github.com/SonarSource/sonar-javascript/blob/34adefdad9ae142cb65159be19e094866574dbc5/javascript-frontend/src/main/java/org/sonar/javascript/cfg/ControlFlowGraphBuilder.java#L238) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1871)

File: javascript-checks/src/main/java/org/sonar/javascript/checks/NonEmptyCaseWithoutBreakCheck.java [hunk 1]
```diff
@@ -110,11 +110,13 @@ public void visitSwitchStatement(SwitchStatementTree tree) {
   }
 
   private static void addCaseExpression(Set<Tree> caseExpressions, ExpressionTree expression) {
-    caseExpressions.add(expression);
-    if (expression.is(Kind.CONDITIONAL_OR)) {
+    if (expression.is(Kind.CONDITIONAL_OR, Kind.CONDITIONAL_AND)) {
       BinaryExpressionTree binary = (BinaryExpressionTree) expression;
-      // The left operand of a "||" is a valid predecessor for switch clause statements
+      // The operands of a "||" or "&&" are valid predecessors for switch clause statements
```
Comment (vilchik-elena) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/NonEmptyCaseWithoutBreakCheck.java]: @pynicolas do we have unit test for this change?

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [This branch's code block is the same as the block for the branch on line 227.](https://github.com/SonarSource/sonar-javascript/blob/6d41a8635399959fb7b0a0ef883695eca51c4e5d/javascript-frontend/src/main/java/org/sonar/javascript/cfg/ControlFlowGraphBuilder.java#L237) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1871)

File: javascript-checks/src/main/java/org/sonar/javascript/checks/NonEmptyCaseWithoutBreakCheck.java (hunk 1 above)
Comment (pynicolas) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/NonEmptyCaseWithoutBreakCheck.java]: @vilchik-elena yes, removing `Kind.CONDITIONAL_AND` from line 118 generates a false positive at [line 56 of the test file](https://github.com/SonarSource/sonar-javascript/blob/dc3ec39f3580d27eb02024f3a59155e0aa3dda6e/javascript-checks/src/test/resources/checks/nonEmptyCaseWithoutBreak.js#L56): it's a `case` preceding `case j && k`.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [This branch's code block is the same as the block for the branch on line 227.](https://github.com/SonarSource/sonar-javascript/blob/f36c74ac1a2a7e0a7a1ccec2f8a2154d35b6c7fc/javascript-frontend/src/main/java/org/sonar/javascript/cfg/ControlFlowGraphBuilder.java#L237) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1871)

**Comment by pynicolas** [comment]
Merged

**Pull request closed without merging**